src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs

```diff
@@ -30,6 +30,27 @@ public void NativeCommandBar_Automated()
 			_app.WaitForText(result, "Clicked!");
 		}
 
+		[Test]
+		[AutoRetry]
+		public void When_Foreground_Is_Set()
+		{
+			Run("UITests.Windows_UI_Xaml_Controls.CommandBar.CommandBar_Native_With_Content");
+			using var screenshot = TakeScreenshot("CommandBar_Native_With_Content.When_Foreground_Is_Set");
+			var rect = _app.Query("MyCommandBar").Single().Rect;
+
+			// Couldn't find a way to get the pixel at the center of the icon. Using this hacky approach:
+			var bitmap = screenshot.GetBitmap();
+			for (int x = (int)rect.Right; x > rect.CenterX; x--)
+			{
+				if (bitmap.GetPixel(x, (int)rect.CenterY) is { A: 255, R: 255, G: 0, B: 0 })
+				{
+					return;
+				}
+			}
+
+			Assert.Fail("Expected a red pixel.");
```

Comment:
I absolutely don't like this approach. But I have tried hard to get it working in a better way and couldn't 😕

Reply:
Can the `AppBarButton` be aligned to the right horizontally? Then you could just sample some pixel with a static offset from the right. But I am not sure what are the paddings applied to the native command bars.

Reply:
If you mean to eliminate the loop, I can do that without changing any alignments. But I'm not sure using a static offset is going to be stable for different screen sizes and resolutions.

Reply:
A static offset is likely not going to work... Let's keep this approach for now, even if it's a blunt.